### PR TITLE
fix: Bluesky PDS template

### PR DESCRIPTION
**Pull request by scanash00**
## Changes
- Added jq and coreutils so psdamin works correctly
- Changed the SERVICE_URL back to SERVICE_FQDN, why was this even changed??!
- Made sure the JWT_SECRET and the PRL_ROTATING_KEY is automatically generated to make it truly one-click deploy
- Added email envs (Important for PDS)
- Made the script shorter by tricking pdsadmin to work using an empty pds.env, as it uses envs from the container

## Issues
- None

### CODE REVIEW

**Comment by scanash00** [comment]
@peaklabs-dev I'd like your approval!

**Comment by madmalkav** [comment]
You need more utils for pdsadmin fully work, see #6605 . Also, the variables PDS_EMAIL_SMTP_URL and PDS_EMAIL_FROM_ADDRESS are missing

**Comment by scanash00** [comment]
@madmalkav there you go, I’ve commited those changes

File: templates/compose/bluesky-pds.yaml [hunk 1]
```diff
@@ -48,6 +48,8 @@ services:
         "PDS_BLOBSTORE_DISK_LOCATION=$${PDS_DATA_DIRECTORY}/blocks" \
         "PDS_BLOB_UPLOAD_LIMIT=$${PDS_BLOB_UPLOAD_LIMIT}" \
         "PDS_DID_PLC_URL=$${PDS_DID_PLC_URL}" \
+        "PDS_EMAIL_FROM_ADDRESS=$${PDS_EMAIL_FROM_ADDRESS}"\
+        "PDS_EMAIL_SMTP_URL=$${PDS_EMAIL_SMTP_URL"}\
```
Comment (mario-neuhold) [inline templates/compose/bluesky-pds.yaml]: Small syntax error:

```suggestion
        "PDS_EMAIL_SMTP_URL=$${PDS_EMAIL_SMTP_URL}"\
```

**Comment by mario-neuhold** [comment]
I had to enter a valid SMTP url here to get the PDS to start successfully:
`- 'PDS_EMAIL_SMTP_URL=${PDS_EMAIL_SMTP_URL}'` = error because PDS_EMAIL_SMTP_URL  does not exist.
`- 'PDS_EMAIL_SMTP_URL=smtps://resend:<API_KEY>@smtp.resend.com:465'` = works

Also, is there a way to get the SMTP URL and EMAIL variables added to the **Service Specific Configuration** sections just like **Admin Password**?

Image

**Comment by madmalkav** [comment]
> I had to enter a valid SMTP url here to get the PDS to start successfully: `- 'PDS_EMAIL_SMTP_URL=${PDS_EMAIL_SMTP_URL}'` = error because PDS_EMAIL_SMTP_URL does not exist. `- 'PDS_EMAIL_SMTP_URL=smtps://resend:<API_KEY>@smtp.resend.com:465'` = works
> 
> Also, is there a way to get the SMTP URL and EMAIL variables added to the **Service Specific Configuration** sections just like **Admin Password**?
> Image

Can you give more information about how exactly it failed? I have a test instance that I deployed with that variable empty with no problems, but there are other changes to the template I'm using there so I'm curious about what can be causing this to you but not to me. I suspect the way @scanash00 generates the pds.env file makes the variable to presnt and empty in the file and that causes problems, but not sure at all that is really the cause.

**Comment by mario-neuhold** [comment]
> Can you give more information about how exactly it failed? I have a test instance that I deployed with that variable empty with no problems, but there are other changes to the template I'm using there so I'm curious about what can be causing this to you but not to me. I suspect the way @scanash00 generates the pds.env file makes the variable to presnt and empty in the file and that causes problems, but not sure at all that is really the cause.

Sure. (Re)starting the ressource shows the info about the missing env:

```
time="2025-09-19T15:53:34+02:00" level=warning msg="The \"PDS_EMAIL_SMTP_URL\" variable is not set. Defaulting to a blank string."
```

And the actual PDS log

```
Generating /pds/pds.env...
Launching PDS...
/app/node_modules/.pnpm/@atproto+pds@0.4.169/node_modules/@atproto/pds/src/config/config.ts:146
      throw new Error(
            ^

Error: Partial email config, must set both emailFromAddress and emailSmtpUrl
    at envToCfg (/app/node_modules/.pnpm/@atproto+pds@0.4.169/node_modules/@atproto/pds/src/config/config.ts:146:13)
    at main (/app/index.js:14:15)
    at Object.<anonymous> (/app/index.js:72:1)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49
```

**Comment by madmalkav** [comment]
Did you set any value for  SERVICE_EMAIL_ADMIN or PDS_EMAIL_FROM_ADDRESS to any value? Can you see what their values are on the environment variable and on the pds.env ?

**Comment by mario-neuhold** [comment]
Yes, I set 
PDS_EMAIL_FROM_ADDRESS = ${SERVICE_EMAIL_ADMIN} 
and
SERVICE_EMAIL_ADMIN to my private email.

The email gets correctly passed to pds.env, I just checked with a debug output. And when I set an SMTP the error goes away.

Here's the whole thing from my "Edit Compose File" with the added email echoing:

```
services:
  pds:
    image: 'ghcr.io/bluesky-social/pds:latest'
    volumes:
      - './pds-data:/pds'
    environment:
      - SERVICE_FQDN_PDS_3000
      - 'PDS_HOSTNAME=${SERVICE_FQDN_PDS_3000}'
      - 'PDS_JWT_SECRET=${SERVICE_HEX_32_JWTSECRET}'
      - 'PDS_ADMIN_PASSWORD=${SERVICE_PASSWORD_ADMIN}'
      - 'PDS_ADMIN_EMAIL=${SERVICE_EMAIL_ADMIN}'
      - 'PDS_PLC_ROTATION_KEY_K256_PRIVATE_KEY_HEX=${SERVICE_HEX_32_ROTATIONKEY}'
      - 'PDS_DATA_DIRECTORY=${PDS_DATA_DIRECTORY:-/pds}'
      - 'PDS_BLOBSTORE_DISK_LOCATION=${PDS_DATA_DIRECTORY:-/pds}/blocks'
      - 'PDS_BLOB_UPLOAD_LIMIT=${PDS_BLOB_UPLOAD_LIMIT:-104857600}'
      - 'PDS_DID_PLC_URL=${PDS_DID_PLC_URL:-https://plc.directory}'
      - 'PDS_EMAIL_FROM_ADDRESS=${SERVICE_EMAIL_ADMIN}'
      - 'PDS_EMAIL_SMTP_URL=${PDS_EMAIL_SMTP_URL}'
      - 'PDS_BSKY_APP_VIEW_URL=${PDS_BSKY_APP_VIEW_URL:-https://api.bsky.app}'
      - 'PDS_BSKY_APP_VIEW_DID=${PDS_BSKY_APP_VIEW_DID:-did:web:api.bsky.app}'
      - 'PDS_REPORT_SERVICE_URL=${PDS_REPORT_SERVICE_URL:-https://mod.bsky.app/xrpc/com.atproto.moderation.createReport}'
      - 'PDS_REPORT_SERVICE_DID=${PDS_REPORT_SERVICE_DID:-did:plc:ar7c4by46qjdydhdevvrndac}'
      - 'PDS_CRAWLERS=${PDS_CRAWLERS:-https://bsky.network}'
      - 'LOG_ENABLED=${LOG_ENABLED:-true}'
    command: |
      sh -c '
        set -euo pipefail; echo "Installing required packages and pdsadmin..."
        apk add --no-cache openssl curl bash jq coreutils gnupg util-linux-misc >/dev/null && \
        curl -o /usr/local/bin/pdsadmin.sh https://raw.githubusercontent.com/bluesky-social/pds/main/pdsadmin.sh && \
        chmod +x /usr/local/bin/pdsadmin.sh && \
        ln -sf /usr/local/bin/pdsadmin.sh /usr/local/bin/pdsadmin

        echo "Generating /pds/pds.env..."
        echo "(debug output) PDS_EMAIL_FROM_ADDRESS value is: $$PDS_EMAIL_FROM_ADDRESS" 
        printf "%s\n" \
        "PDS_HOSTNAME=$${PDS_HOSTNAME}" \
        "PDS_JWT_SECRET=$${PDS_JWT_SECRET}" \
        "PDS_ADMIN_PASSWORD=$${PDS_ADMIN_PASSWORD}" \
        "PDS_ADMIN_EMAIL=$${PDS_ADMIN_EMAIL}" \
        "PDS_PLC_ROTATION_KEY_K256_PRIVATE_KEY_HEX=$${PDS_PLC_ROTATION_KEY_K256_PRIVATE_KEY_HEX}" \
        "PDS_DATA_DIRECTORY=$${PDS_DATA_DIRECTORY}" \
        "PDS_BLOBSTORE_DISK_LOCATION=$${PDS_DATA_DIRECTORY}/blocks" \
        "PDS_BLOB_UPLOAD_LIMIT=$${PDS_BLOB_UPLOAD_LIMIT}" \
        "PDS_DID_PLC_URL=$${PDS_DID_PLC_URL}" \
        "PDS_EMAIL_FROM_ADDRESS=$${PDS_EMAIL_FROM_ADDRESS}"\
        "PDS_EMAIL_SMTP_URL=$${PDS_EMAIL_SMTP_URL}"\
        "PDS_BSKY_APP_VIEW_URL=$${PDS_BSKY_APP_VIEW_URL}" \
        "PDS_BSKY_APP_VIEW_DID=$${PDS_BSKY_APP_VIEW_DID}" \
        "PDS_REPORT_SERVICE_URL=$${PDS_REPORT_SERVICE_URL}" \
        "PDS_REPORT_SERVICE_DID=$${PDS_REPORT_SERVICE_DID}" \
        "PDS_CRAWLERS=$${PDS_CRAWLERS}" \
        "LOG_ENABLED=$${LOG_ENABLED}" \
        > /pds/pds.env

        echo "Launching PDS..."
        exec node --enable-source-maps index.js
      '
    healthcheck:
      test:
        - CMD
        - wget
        - '--spider'
        - 'http://127.0.0.1:3000/xrpc/_health'
      interval: 2s
      timeout: 10s
      retries: 10
```

**Comment by madmalkav** [comment]
PDS wants PDS_EMAIL_FROM_ADDRESS and PDS_EMAIL_SMTP_URL both empty or both filled. If you have no intention of sending emails, I’ll recommend leaving both blank and set only PDS_ADMIN_EMAILEl 19 sept 2025, a las 16:27, Mario Neuhold ***@***.***> escribió:﻿mario-neuhold left a comment (coollabsio/coolify#6636)
Yes, I set
PDS_EMAIL_FROM_ADDRESS = ${SERVICE_EMAIL_ADMIN}
and
SERVICE_EMAIL_ADMIN to my private email.
The email gets correctly passed to pds.env, I just checked with a debug output. And when I set an SMTP the error goes away.
Here's the whole thing from mz "Edit Compose File" with the added email echoing:
services:
  pds:
    image: 'ghcr.io/bluesky-social/pds:latest'
    volumes:
      - './pds-data:/pds'
    environment:
      - SERVICE_FQDN_PDS_3000
      - 'PDS_HOSTNAME=${SERVICE_FQDN_PDS_3000}'
      - 'PDS_JWT_SECRET=${SERVICE_HEX_32_JWTSECRET}'
      - 'PDS_ADMIN_PASSWORD=${SERVICE_PASSWORD_ADMIN}'
      - 'PDS_ADMIN_EMAIL=${SERVICE_EMAIL_ADMIN}'
      - 'PDS_PLC_ROTATION_KEY_K256_PRIVATE_KEY_HEX=${SERVICE_HEX_32_ROTATIONKEY}'
      - 'PDS_DATA_DIRECTORY=${PDS_DATA_DIRECTORY:-/pds}'
      - 'PDS_BLOBSTORE_DISK_LOCATION=${PDS_DATA_DIRECTORY:-/pds}/blocks'
      - 'PDS_BLOB_UPLOAD_LIMIT=${PDS_BLOB_UPLOAD_LIMIT:-104857600}'
      - 'PDS_DID_PLC_URL=${PDS_DID_PLC_URL:-https://plc.directory}'
      - 'PDS_EMAIL_FROM_ADDRESS=${SERVICE_EMAIL_ADMIN}'
      - 'PDS_EMAIL_SMTP_URL=${PDS_EMAIL_SMTP_URL}'
      - 'PDS_BSKY_APP_VIEW_URL=${PDS_BSKY_APP_VIEW_URL:-https://api.bsky.app}'
      - 'PDS_BSKY_APP_VIEW_DID=${PDS_BSKY_APP_VIEW_DID:-did:web:api.bsky.app}'
      - 'PDS_REPORT_SERVICE_URL=${PDS_REPORT_SERVICE_URL:-https://mod.bsky.app/xrpc/com.atproto.moderation.createReport}'
      - 'PDS_REPORT_SERVICE_DID=${PDS_REPORT_SERVICE_DID:-did:plc:ar7c4by46qjdydhdevvrndac}'
      - 'PDS_CRAWLERS=${PDS_CRAWLERS:-https://bsky.network}'
      - 'LOG_ENABLED=${LOG_ENABLED:-true}'
    command: |
      sh -c '
        set -euo pipefail; echo "Installing required packages and pdsadmin..."
        apk add --no-cache openssl curl bash jq coreutils gnupg util-linux-misc >/dev/null && \
        curl -o /usr/local/bin/pdsadmin.sh https://raw.githubusercontent.com/bluesky-social/pds/main/pdsadmin.sh && \
        chmod +x /usr/local/bin/pdsadmin.sh && \
        ln -sf /usr/local/bin/pdsadmin.sh /usr/local/bin/pdsadmin

        echo "Generating /pds/pds.env..."
        echo "(debug output) PDS_EMAIL_FROM_ADDRESS value is: $$PDS_EMAIL_FROM_ADDRESS" 
        printf "%s\n" \
        "PDS_HOSTNAME=$${PDS_HOSTNAME}" \
        "PDS_JWT_SECRET=$${PDS_JWT_SECRET}" \
        "PDS_ADMIN_PASSWORD=$${PDS_ADMIN_PASSWORD}" \
        "PDS_ADMIN_EMAIL=$${PDS_ADMIN_EMAIL}" \
        "PDS_PLC_ROTATION_KEY_K256_PRIVATE_KEY_HEX=$${PDS_PLC_ROTATION_KEY_K256_PRIVATE_KEY_HEX}" \
        "PDS_DATA_DIRECTORY=$${PDS_DATA_DIRECTORY}" \
        "PDS_BLOBSTORE_DISK_LOCATION=$${PDS_DATA_DIRECTORY}/blocks" \
        "PDS_BLOB_UPLOAD_LIMIT=$${PDS_BLOB_UPLOAD_LIMIT}" \
        "PDS_DID_PLC_URL=$${PDS_DID_PLC_URL}" \
        "PDS_EMAIL_FROM_ADDRESS=$${PDS_EMAIL_FROM_ADDRESS}"\
        "PDS_EMAIL_SMTP_URL=$${PDS_EMAIL_SMTP_URL}"\
        "PDS_BSKY_APP_VIEW_URL=$${PDS_BSKY_APP_VIEW_URL}" \
        "PDS_BSKY_APP_VIEW_DID=$${PDS_BSKY_APP_VIEW_DID}" \
        "PDS_REPORT_SERVICE_URL=$${PDS_REPORT_SERVICE_URL}" \
        "PDS_REPORT_SERVICE_DID=$${PDS_REPORT_SERVICE_DID}" \
        "PDS_CRAWLERS=$${PDS_CRAWLERS}" \
        "LOG_ENABLED=$${LOG_ENABLED}" \
        > /pds/pds.env

        echo "Launching PDS..."
        exec node --enable-source-maps index.js
      '
    healthcheck:
      test:
        - CMD
        - wget
        - '--spider'
        - 'http://127.0.0.1:3000/xrpc/_health'
      interval: 2s
      timeout: 10s
      retries: 10

**Comment by mario-neuhold** [comment]
Oh, you are absolutely correct - thank you for that hint. I can confirm PDS is starting up just fine when both variables are not set.
**But** leaving them empty is not enough, I have to remove them completely.

**Comment by madmalkav** [comment]
Can you please try this version of the template? I did a superquick test and it should be OK but couldn't test as much as I would like...

If you care about the PDS you are creating, remember to keep the keys generated for it.

https://github.com/madmalkav/coolify/blob/v4.x/templates/compose/bluesky-pds.yaml

**Comment by scanash00** [comment]
> Can you please try this version of the template? I did a superquick test and it should be OK but couldn't test as much as I would like...
> 
> If you care about the PDS you are creating, remember to keep the keys generated for it.
> 
> https://github.com/madmalkav/coolify/blob/v4.x/templates/compose/bluesky-pds.yaml

I’ll try it when I can!

**Comment by mario-neuhold** [comment]
> Can you please try this version of the template? I did a superquick test and it should be OK but couldn't test as much as I would like...
> 
> If you care about the PDS you are creating, remember to keep the keys generated for it.
> 
> https://github.com/madmalkav/coolify/blob/v4.x/templates/compose/bluesky-pds.yaml

That one is working fine. I gotta admit, I might have had a lot of leftover env baggage in my ressource earlier since I was only changing the docker-compose via "Edit Compose File" from the initial setup with the default bsky-pds template, then the one from this PR from @scanash00 and then yours @madmalkav and doing some manual edits in between.

Now I tried adding a fresh resource with your template as a plain docker-compose and the PDS is running without having to enter anything in the environment variables.

But the one from this PR is also working fine in a fresh ressource. It was really just a mess on my side, sorry about all the fuzz I created and thanks a lot for your help!

**Comment by scanash00** [comment]
@madmalkav thank you again!

**Comment by scanash00** [comment]
I went ahead and made both email envs required, so they won't be able to deploy their pds without checking both, and they should know how to remove the envs from the compose if they don't want to add email configuration to their pds, which is not recommended

**Comment by scanash00** [comment]
@coderabbitai what do you think of this boss

**Comment by madmalkav** [comment]
> I went ahead and made both email envs required, so they won't be able to deploy their pds without checking both, and they should know how to remove the envs from the compose if they don't want to add email configuration to their pds, which is not recommended

To be honest I don't like this idea, required variables should be, well, the ones that are truly required. What you propose is not very intuitive. Also, what is the volumes at the last two lines?

I think it would be better to close this PR and make a new one from my last version, as it integrates all the fixes discussed here and has been tested by several people.

**Comment by scanash00** [comment]
> > I went ahead and made both email envs required, so they won't be able to deploy their pds without checking both, and they should know how to remove the envs from the compose if they don't want to add email configuration to their pds, which is not recommended
> 
> To be honest I don't like this idea, required variables should be, well, the ones that are truly required. What you propose is not very intuitive. Also, what is the volumes at the last two lines?
> 
> I think it would be better to close this PR and make a new one from my last version, as it integrates all the fixes discussed here and has been tested by several people.

@madmalkav The volumes make sure it does it in a docker volume instead of root. And your template is similar to mine with less fixes.

**Comment by scanash00** [comment]
And both email envs are required at the start because it won’t start with both of them being filled or deleted.

**Comment by madmalkav** [comment]
- I still don't understand the following on lines 54 and 55

```
volumes:
  pds-data:
```

- I also don't understand your answer about PDS_EMAIL_FROM_ADDRESS and PDS_EMAIL_SMTP_URL. They can be blank, PDS software doesn't enforce it and works fine without it, and I still think it is wrong to mark as required parameters that aren't mandatory to run the container. Many people running a single user PDS don't set this configuration.
- you haven't changed the blob size. Now Bluesky accepts 100MB videos, by default a PDS installed with this template won't be able to upload a 100MB video.
- You changed back SERVICE_URL_PDS_3000 to SERVICE_FQDN_PDS_3000, so the link on the Coolify UI is wrong
imagen

**Comment by scanash00** [comment]
> * I still don't understand the following on lines 54 and 55
> 
> ```
> volumes:
>   pds-data:
> ```
> 
>  * I also don't understand your answer about PDS_EMAIL_FROM_ADDRESS and PDS_EMAIL_SMTP_URL. They can be blank, PDS software doesn't enforce it and works fine without it, and I still think it is wrong to mark as required parameters that aren't mandatory to run the container. Many people running a single user PDS don't set this configuration.
> * you haven't changed the blob size. Now Bluesky accepts 100MB videos, by default a PDS installed with this template won't be able to upload a 100MB video.
> * You changed back SERVICE_URL_PDS_3000 to SERVICE_FQDN_PDS_3000, so the link on the Coolify UI is wrong
> 
> imagen

first of all I did change the blob size.

the volumes are set so the PDS data is saved on reboots

**Comment by madmalkav** [comment]
Ouch, my bad about the blob size, too many open windows 😂

I still don't understand having volumes declared twice at the beginning and the end of the file, but its probably a lack in my understanding of Docker

**Comment by scanash00** [comment]
> Ouch, my bad about the blob size, too many open windows 😂
> 
> I still don't understand having volumes declared twice at the beginning and the end of the file, but its probably a lack in my understanding of Docker

Alright, so the first volume declaration is where you declare where it's actually mounted, the second one if where you tell docker the configurations for it.

I am pretty sure the bottom one makes sure Coolify makes a /var/lib/docker/volumes directory for it?

**Comment by scanash00** [comment]
CleanShot 2025-09-20 at 11 41 27@2x
Yeah it makes a directory on /var/lib/docker/volumes for it.

**Comment by madmalkav** [comment]
And why you want to do that instead of having it on the /data/coolify/services like the other services do?

**Comment by scanash00** [comment]
> And why you want to do that instead of having it on the /data/coolify/services like the other services do?

That's where the compose of those services is stored, not the data itself

**Comment by scanash00** [comment]
My docker and coolify knowledge improved throughout this PR

**Comment by scanash00** [comment]
I think this can be merged anytime now

**Comment by scanash00** [comment]
@andrasbacsai @peaklabs-dev This is ready to merge 👀 fixes a lot of things wrong with the template don't let it die!

**Comment by NeilHanlon** [comment]
If someone wanted to test this PR, how would they do that? Any simple instructions?

**Comment by scanash00** [comment]
CleanShot 2025-10-01 at 22 53 09@2x
@NeilHanlon You would copy the compose from the Files Changed tab and paste it here

**Comment by NeilHanlon** [comment]
@scanash00 Thanks! Wasn't sure if there was some magic going on behind the scenes.

**Comment by NeilHanlon** [comment]
This works great! 👍🏼

**Comment by joshfarrant** [comment]
I'm keen to test this too as I'm learning more about running a PDS with Coolify, and this change certainly gets me further than the existing PDS template, so it seems like a marked improvement to me 👍

Excuse my ignorance here as I think I'm probably missing something fundamental. Once I've got this PDS running, how do I create an account which I can then use to auth on bsky.app?

**Comment by scanash00** [comment]
@joshfarrant Hey,

You would need to run: pdsadmin create-invite-code and copy that code, then you would go to bsky.app, change where it says “creating an account on bluesky social” select “Custom” and enter the url of your PDS, it will ask you to enter that invite code, and then register as usual.

**Comment by joshfarrant** [comment]
Hey @scanash00, thanks for the quick response!


I tried the following from the Coolify service's terminal

```sh
/usr/local/bin/pdsadmin create-invite-code
```

Which resulted in the following,

```
curl: (60) SSL certificate problem: self-signed certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the webpage mentioned above.
```

**Comment by scanash00** [comment]
@joshfarrant yeah, that happens when you are not using https for your instance. Try using a custom domain with https!

**Comment by joshfarrant** [comment]
Hmm, I am using https 🤔 https://pds.farrant.me

**Comment by scanash00** [comment]
Bad missclick lol

**Comment by scanash00** [comment]
@joshfarrant interesting. Can you check your envs and see if they are right?

**Comment by joshfarrant** [comment]
@scanash00 You got it 💖 My `SERVICE_FQDN_PDS_3000` was still set to the `sslip.io` domain for some reason. Really appreciate the help, thanks!

**Comment by scanash00** [comment]
I’m on a long flight ✈️ I’ll do it whenever i am ready

**Comment by scanash00** [comment]
Bluesky PDS sellf-hosting has gained popularity this weekend!

**Comment by scanash00** [comment]
Because this is taking a bit too get merged, I made a video on how to get it setup!


https://github.com/user-attachments/assets/bbf3bfc6-2701-44b1-a692-4574c822cf4b

File: templates/compose/bluesky-pds.yaml [hunk 2]
```diff
@@ -9,59 +9,45 @@ services:
   pds:
     image: 'ghcr.io/bluesky-social/pds:latest'
```
Comment (ShadowArcanist) [inline templates/compose/bluesky-pds.yaml]: Please pin the version to a static version instead of using `latest` so the template won't stop working if they introduce some changes to latest version

File: templates/compose/bluesky-pds.yaml [hunk 3]
```diff
@@ -9,59 +9,45 @@ services:
...
     volumes:
-      - ./pds-data:/pds
+      - './pds-data:/pds'
```
Comment (ShadowArcanist) [inline templates/compose/bluesky-pds.yaml]:
```suggestion
      - 'pds-data:/pds'
```

**Review by ShadowArcanist — CHANGES_REQUESTED** [review]
Hey @scanash00 , I checked this PR and the service works fine for me, could you please implement the changes I requested below

File: templates/compose/bluesky-pds.yaml (hunk 3 above)
Comment (scanash00) [reply templates/compose/bluesky-pds.yaml]: No need to, it’s fine.

**Review by ShadowArcanist — APPROVED** [review]

File: templates/compose/bluesky-pds.yaml [hunk 4]
```diff
@@ -7,61 +7,47 @@
 
 services:
   pds:
-    image: 'ghcr.io/bluesky-social/pds:latest'
+    image: 'ghcr.io/bluesky-social/pds:0.4.182'
     volumes:
-      - ./pds-data:/pds
+      - './pds-data:/pds'
```
Comment (Cinzya) [inline templates/compose/bluesky-pds.yaml]: will have to insist making this a volume instead of bind mount, unless there is a specific reason for it. There are often bugs in coolify that cause the permissions for the `/data` directory to be overriden. This will make every app with bind mounts on the root stop working,
```suggestion
      - pds-data:/pds
```

**Review by Cinzya — CHANGES_REQUESTED** [review]

File: templates/compose/bluesky-pds.yaml (hunk 2 above)
Comment (madmalkav) [reply templates/compose/bluesky-pds.yaml]: Given a new release is made each other month aprox, maybe waiting for the template to be updated each time is a tad too much? Can I suggest a middle term of using tag 0.4 as I doubt any breaking change would be introduced without a version bump?

PS: I don't know if this suggestion is OK or against Coolify template standards, maybe https://coolify.io/docs/get-started/contribute/service#:~:text=Add%20a%20new%20service%20template%20to%20Coolify can add some guidelines about choosing versions?

Comment (scanash00) [reply templates/compose/bluesky-pds.yaml]: @madmalkav users can update their PDS themselves, it’s set the latest PDS version right now which works with the template

**Comment by joshfarrant** [comment]
Excuse another potentially obvious question, but is there a straightforward way to back up a PDS created using this docker-compose? In some Coolify services — say Umami — I see a 'Backups' option which I can point at my bucket of choice and configure automatic backups, but that appears to be missing for this PDS service.

Is that something which can be configured via the docker-compose, and would that be resolved by implementing [the above comment](https://github.com/coollabsio/coolify/pull/6636#discussion_r2413203672)?

**Comment by scanash00** [comment]
> Excuse another potentially obvious question, but is there a straightforward way to back up a PDS created using this docker-compose? In some Coolify services — say Umami — I see a 'Backups' option which I can point at my bucket of choice and configure automatic backups, but that appears to be missing for this PDS service.
> 
> Is that something which can be configured via the docker-compose, and would that be resolved by implementing [the above comment](https://github.com/coollabsio/coolify/pull/6636#discussion_r2413203672)?

@joshfarrant that comment was implemented.


You can back up your database by going to the /var/lib/docker/{your_volume_name} and backup that.

**Comment by andrasbacsai** [comment]
> Excuse another potentially obvious question, but is there a straightforward way to back up a PDS created using this docker-compose? In some Coolify services — say Umami — I see a 'Backups' option which I can point at my bucket of choice and configure automatic backups, but that appears to be missing for this PDS service.
> 
> Is that something which can be configured via the docker-compose, and would that be resolved by implementing [the above comment](https://github.com/coollabsio/coolify/pull/6636#discussion_r2413203672)?

I will also introduce docker volume backups like we do with databases soon.

**Comment by andrasbacsai** [comment]
Thank you for the PR!
